storage/google/cloud/storage/blob.py

```diff
@@ -503,8 +503,13 @@ def download_to_filename(self, filename, client=None):
 
         :raises: :class:`google.cloud.exceptions.NotFound`
         """
-        with open(filename, 'wb') as file_obj:
-            self.download_to_file(file_obj, client=client)
+        try:
+            with open(filename, 'wb') as file_obj:
+                self.download_to_file(file_obj, client=client)
+        except resumable_media.DataCorruption as exc:
+            # Delete the corrupt downloaded file.
+            os.remove(filename)
+            raise
```

Comment:
@mfschwartz Can you take a peek at this change? It is different in two ways from your original change:

- It is in `download_to_filename` (vs. `download_to_file`). This method is the only method where we **know** we have created the file, so it is safe to delete it. In `download_to_file`, the user could pass in a non-file but file-like object, so there would be nothing to delete. (It should really be named `download_to_stream`, but we are already 1.0 so yay!)
- I am just calling `raise` to re-raise the error. The previous approach `raise exceptions.from_http_status` didn't make much sense because the status code would be 200 or 206 (and I think it would just be a 200). We don't have any corresponding error class, so it'd just [be][1] a generic `GoogleAPICallError`

[1]: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/5cc463f934f165fc8685725d9f0b60a1043dd50d/core/google/api/core/exceptions.py#L337

Reply:
LGTM. Thanks for those fixes.